interactive_engine/executor/ir/runtime/src/process/entry.rs

```diff
@@ -0,0 +1,619 @@
+                EntryDataType::P => self.as_graph_path().eq(&other.as_graph_path()),
+                EntryDataType::Obj => self.as_object().eq(&other.as_object()),
+                EntryDataType::Intersect => self
+                    .as_any_ref()
+                    .downcast_ref::<Intersection>()
+                    .eq(&other
+                        .as_any_ref()
+                        .downcast_ref::<Intersection>()),
+                EntryDataType::Collection => self
+                    .as_any_ref()
+                    .downcast_ref::<CollectionEntry>()
+                    .eq(&other
+                        .as_any_ref()
+                        .downcast_ref::<CollectionEntry>()),
+            }
+        } else {
+            false
+        }
+    }
+}
+
+// demanded when need to  order the entry;
+impl PartialOrd for DynEntry {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        if (self.get_type()).eq(&other.get_type()) {
+            match self.get_type() {
+                EntryDataType::Vid | EntryDataType::V => self.as_vid().partial_cmp(&other.as_vid()),
+                EntryDataType::E => self.as_edge().partial_cmp(&other.as_edge()),
+                EntryDataType::P => self
+                    .as_graph_path()
+                    .partial_cmp(&other.as_graph_path()),
+                EntryDataType::Obj => self.as_object().partial_cmp(&other.as_object()),
+                EntryDataType::Intersect => self
+                    .as_any_ref()
+                    .downcast_ref::<Intersection>()
+                    .partial_cmp(
+                        &other
+                            .as_any_ref()
+                            .downcast_ref::<Intersection>(),
+                    ),
+                EntryDataType::Collection => self
+                    .as_any_ref()
+                    .downcast_ref::<CollectionEntry>()
+                    .partial_cmp(
+                        &other
+                            .as_any_ref()
+                            .downcast_ref::<CollectionEntry>(),
+                    ),
+            }
+        } else {
+            None
+        }
+    }
+}
+
+// demanded when need to group (ToSet) the entry;
+impl Eq for DynEntry {}
+
+// demanded when need to group (ToSum) the entry;
+impl std::ops::Add for DynEntry {
+    type Output = DynEntry;
+
+    fn add(self, rhs: Self) -> Self::Output {
+        if (self.get_type()).eq(&rhs.get_type()) {
+            if EntryDataType::Obj.eq(&self.get_type()) {
+                match (self.as_object(), rhs.as_object()) {
+                    (Some(Object::Primitive(p1)), Some(Object::Primitive(p2))) => {
+                        return DynEntry::new(Object::Primitive(p1.add(p2.clone())));
+                    }
+                    _ => {}
+                }
+            }
+        }
+        DynEntry::new(Object::None)
+    }
+}
+
+#[derive(Debug, Clone, Default)]
+pub struct IDEntry {
```

Comment:
`IdEntry`. BTW, for the next a few definitions, can put `VertexEntry`, `EdgeEntry`, `PathEntry`, `IntersectionEntry` and `CollectionEntry`.  Anyway, if we already have `DynEntry`, why we should have so many different entry definitions?